chainer_chemistry/dataset/splitters/stratified_splitter.py

```diff
@@ -0,0 +1,212 @@
+import numpy
+import pandas
+
+from chainer_chemistry.dataset.splitters.base_splitter import BaseSplitter
+from chainer_chemistry.datasets.numpy_tuple_dataset import NumpyTupleDataset
+
+
+def _approximate_mode(class_counts, n_draws):
+    n_class = len(class_counts)
+    continuous = class_counts * n_draws / class_counts.sum()
+    floored = numpy.floor(continuous)
+    assert n_draws // n_class == floored.sum() // n_class
+    n_remainder = int(n_draws - floored.sum())
+    remainder = continuous - floored
+    inds = numpy.argsort(remainder)[::-1]
+    inds = inds[:n_remainder]
+    floored[inds] += 1
+    assert n_draws == floored.sum()
+    return floored.astype(numpy.int)
+
+
+class StratifiedSplitter(BaseSplitter):
+    """Class for doing stratified data splits."""
+
+    def _split(self, dataset, frac_train=0.8, frac_valid=0.1, frac_test=0.1,
+               labels=None, **kwargs):
+        numpy.testing.assert_almost_equal(frac_train + frac_valid + frac_test,
+                                          1.)
+
+        seed = kwargs.get('seed', None)
+        label_axis = kwargs.get('label_axis', -1)
+        task_index = kwargs.get('task_index', 0)
+        n_bin = kwargs.get('n_bin', 10)
+        task_type = kwargs.get('task_type', 'infer')
+        if task_type not in ['classification', 'regression', 'infer']:
+            raise ValueError("{} is invalid. Please use 'classification',"
+                             "'regression' or 'infer'".format(task_type))
+
+        rng = numpy.random.RandomState(seed)
+
+        if labels is None:
+            if not isinstance(dataset, NumpyTupleDataset):
+                raise ValueError("Please assign label dataset.")
+            labels = dataset.features[:, label_axis]
+
+        if len(labels.shape) == 1:
+            labels = labels
+        else:
+            labels = labels[:, task_index]
+
+        if task_type == 'infer':
+            if labels.dtype.kind == 'i':
+                task_type = 'classification'
+            elif labels.dtype.kind == 'f':
+                task_type = 'regression'
+            else:
+                raise ValueError
+
+        if task_type == 'classification':
+            classes, labels = numpy.unique(labels, return_inverse=True)
+        elif task_type == 'regression':
+            classes = numpy.arange(n_bin)
+            labels = pandas.qcut(labels, n_bin, labels=False)
+        else:
+            raise ValueError
+
+        n_classes = classes.shape[0]
+        n_total_valid = int(numpy.floor(frac_valid * len(dataset)))
+        n_total_test = int(numpy.floor(frac_test * len(dataset)))
```

Comment:
please put comment
\# n_total_train is the remainder: n - n_total_valid - n_total_test

Reply:
I wrote comments in other place.